application-ckeditor-ui/src/main/resources/CKEditor/EditSheet.xml

```diff
@@ -390,6 +400,29 @@ define('xwiki-ckeditor', [
     return oldInline.call(this, element, ckeditor.tools.extend(getConfig(element), config, true));
   };
 
+  // Define a way to switch on or off the temporary attachment upload mechanism.
+  ckeditor.switchTemporaryAttachmentUpload = function (isEnabled, instance, config) {
+    if (!config) {
+      var config = instance.config;
+    }
+    var element = instance.element;
+    if (!config['xwiki-upload']) {
+      config['xwiki-upload'] = {
+        'isTemporaryAttachmentSupported': $(element).attr('data-isTemporaryAttachmentSupported')
+      }
+    }
+
+    if (isEnabled !== config['xwiki-upload'].isTemporaryAttachmentSupported) {
+      var sourceDocument = getDocument(element);
+      config.filebrowserUploadUrl = getUploadURL(sourceDocument, 'filebrowser', isEnabled);
+      config.uploadUrl = getUploadURL(sourceDocument, 'filetools', isEnabled);
+      config['xwiki-upload'].isTemporaryAttachmentSupported = isEnabled;
+      $(element).attr('data-isTemporaryAttachmentSupported', isEnabled);
+      // Those properties should never be overridden by the in-page configuration.
+      ckeditor.keepConfigValues = ['xwiki-upload', 'filebrowserUploadUrl', 'uploadUrl'];
```

Comment:
Why? I don't see why they shouldn't. This looks hackish.

Reply:
This is a bit hackish, it was to make working the disabling of the feature in the advanced editor configuration in XWiki. But as I said in a previous comment, maybe that's not needed.